game-core/src/main/java/games/strategy/triplea/ai/pro/util/MoveBatcher.java

```diff
@@ -0,0 +1,161 @@
+package games.strategy.triplea.ai.pro.util;
+
+import games.strategy.engine.data.Route;
+import games.strategy.engine.data.Unit;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * MoveBatcher implements an algorithm that batches AI moves together, reducing the number of
+ * distinct moves the AI does (and thus the visual delay) by combining moves together.
+ */
+public class MoveBatcher {
+  private static class Move {
+    private final ArrayList<Unit> units;
+    private final Route route;
+    private final @Nullable HashMap<Unit, Unit> unitsToTransports;
+
+    private Move(
+        final ArrayList<Unit> units,
+        final Route route,
+        final @Nullable HashMap<Unit, Unit> unitsToTransports) {
+      this.units = units;
+      this.route = route;
+      this.unitsToTransports = unitsToTransports;
+    }
+
+    Move(final ArrayList<Unit> units, final Route route) {
+      this(units, route, null);
+    }
+
+    Move(final Unit unit, final Route route, final Unit transportToLoad) {
+      this(new ArrayList<>(List.of(unit)), route, new HashMap<>(Map.of(unit, transportToLoad)));
+    }
+
+    private boolean isTransportLoad() {
```

Comment:
It means - "is the type of this move a transport load"? i.e. as part of this move, units are being loaded on a transport.

So none of your above names apply.

We can name it isTransportLoadMove() - but this a method of a class called Move, so the Move at the end is redundant - it's like if List.isEmpty() was called List.isEmptyList() - sure it's correct, but kind of redundant.

Perhaps adding Javadoc would be sufficient for you here?

Other options are - isMoveTypeTransportLoad() - or isLoadingUnits()... Not sure I like them better than current name if you're OK with just Javadoc.

Reply:
Maybe `isTransporting`?
A private method probably should be self-explanatory through it's name, I feel like javadoc'ing a one-liner private method means we have not come up with a good name/abstraction.

Reply:
isTransporting() sounds like it's a "move of units together with transport" not a move to load units onto a transport. I assume the confusion is with the word "load" since it can mean different things. Do we want different terminology? IsEmbarkation()? But then we're introducing new terminology which might be more confusing.

Reply:
`unitsAreLoadingOntoTransport()`?
Failing that, yeah, a quick javadoc just in case.